internal/filedata/archive_manager.go

```diff
@@ -100,106 +88,42 @@ func (am *ArchiveManager) Close() error {
-				if err != nil {
-					// A failure here might be a real failure, or it might be that the file is being copied
-					// over non-atomically so that we're seeing an invalid partial state. So we'll always
-					// retry at least once in this case.
-					am.loggers.Warnf(logMsgReloadError, err.Error())
-					lastError = err
-					scheduleRetry()
-					return
-				}
-				am.loggers.Warnf(logMsgReloadedData, am.filePath)
-				am.updatedArchive(ar)
-				ar.Close()
-				return
-			}
-			am.loggers.Debugf("File has not changed (size=%d, mtime=%s)", curFileInfo.Size(), curFileInfo.ModTime())
-			if lastError == nil {
-				// This was a spurious file watch notification - the file hasn't changed and we're not retrying
-				// after an error, so there's nothing to do
-				return
-			}
-			am.loggers.Warn(logMsgReloadUnchangedRetry)
-		} else if lastError == nil {
-			am.loggers.Warn(logMsgReloadFileNotFound)
-			lastError = err
-		}
-		// If we got here, then either the file was not found, or we triggered a delayed retry after
-		// an earlier error and the file has not changed since the last failed attempt.
-		//
-		// So there's no point in trying to reload it now, but it's still possible that there's a slow
-		// copy operation in progress, so we'll schedule another retry-- up to a limit. We won't rely on
-		// the file watching mechanism for this, because its granularity might be too large to detect
-		// consecutive changes that happen close together.
-		if firstRetryTime.IsZero() || time.Since(firstRetryTime) < maxRetryDuration {
-			scheduleRetry()
-		} else {
-			am.loggers.Errorf(logMsgReloadUnchangedNoMoreRetries, lastError)
-		}
-	}
+	am.loggers.Infof("Monitoring %s for changes (every %s) (size=%d, mtime=%s)", am.filePath, am.monitoringInterval, original.Size(), original.ModTime())
 
 	for {
 		select {
 		case <-am.closeCh:
-			_ = am.watcher.Close()
 			return
-
-		case event := <-am.watcher.Events:
-			am.loggers.Debugf("Got file watcher event: %+v", event)
-			// If we are already going to retry after a brief interval, then we can ignore any file watch
-			// events that are triggered before the retry. Some implementations of fsnotify can produce a
-			// burst of redundant events, and there's no point in trying to reload the file many times
-			// within our retry interval.
-			if pendingRetry {
-				am.loggers.Debug("Ignoring file watcher event because there is already a scheduled retry")
-			} else {
-				maybeReload()
+		case <-ticker.C:
+			nextInfo, err := os.Stat(am.filePath)
+			if err != nil {
+				if os.IsNotExist(err) {
+					am.loggers.Errorf("File %s not found", am.filePath)
+				} else {
+					am.loggers.Errorf("Error: %s", err)
+				}
+				continue
 			}
-
-		case <-retryCh:
-			// If needRetry is false, this is an obsolete signal - we've already successfully reloaded
-			if pendingRetry {
-				am.loggers.Debug("Got retry signal")
-				pendingRetry = false
-				maybeReload()
+			if fileMayHaveChanged(prevInfo, nextInfo) {
```

Comment:
So, one thing to note, this `fileMayhaveChanged` would be a batch match for use with a watcher. If you programmatically modify a file, then it gets several changes in a small period. This period is less than the timestamp precision, which is 1 second typically. So depending on how things go you get random states within that update cycle. Node server had this problem.

I do think there is a letter problem here. Which is 100ms is also faster, multiple updates within 1 second you could load the first change, and then subsequently thetimestamp remains the same, so you don't. But it isn't too bad because it also has to be the same size. So it would affect a change of variation from something like "a" to "b".